---
title: Detailed programme
---

# Functional analysis 

Mark Webber and Cynthia Whitchurch

## Week 6 Beginning 12 December 2022 

_Functional analysis: how can we understand what does what?_

As you have seen in the previous weeks, the ability to generate large amounts of bacterial genomic sequence data is now a routine activity\. Public health agencies and research laboratories have now generated hundreds of thousands of genomes for many pathogens \(for an example see the enterobase database: [https://enterobase\.warwick\.ac\.uk/](https://enterobase.warwick.ac.uk/)\)\. However, whilst this explosion in sequence data has allowed us to document a lot of ‘what’s there?’, a central problem remains in understanding what all this DNA does\. 

In this week we will explore some of the methods available to understand the relationships between genotype and phenotype\. We will follow a common theme through the week – exploring how different methods can help understand how some important bacterial pathogens respond to a major antibiotic\.

We will introduce and apply different methodologies to understand how different expression of genes or proteins or changes in bacterial behaviors can impact survival of important pathogens to antibiotics\. We will ask you to integrate date from different methods to formulate hypotheses about how pathogens can survive antibiotic exposure\.

A useful primer to mechanisms of AMR can be found here: [https://www\.nature\.com/articles/s41579\-022\-00820\-y](https://www.nature.com/articles/s41579-022-00820-y) 

Datasets to use will be derived from both *E\. coli* and *P\. aeruginosa* after exposure to the important carbapenem antibiotic, meropenem \([https://go\.drugbank\.com/drugs/DB00760](https://go.drugbank.com/drugs/DB00760)\) and will include

- RNAseq 
- TraDIS
- Morphological observations

Structure of the week:

* Monday: Image analysis
* Tuesday: RNAseq data
* Thursday: TraDIS
* Friday: Hypothesis building and wrap up

### Monday – Introduction and importance of bacterial behaviour

Not all adaptation requires mutation – some bacterial behaviours \(changes in morphology, formation of a biofilm etc\) can impact their ability to survive without mutation or gain of any DNA\. This can be observed using sophisticated microscopy and data can be extracted which allows powerful analysis\. We will demonstrate how image data can be used to understand bacterial stress adaptations and explain some of the informatics approaches used\.

*Schedule*

- Introduction to the week ahead – Mark Webber\. 9\.30\-10
- Image analysis tutorial – James Lazenby/ Cynthia Whitchurch\. 10\-12
- Lunch\. 12\-1
- Image analysis continued \- James Lazenby/ Cynthia Whitchurch\. 1\-close

### Tuesday – how gene expression can identify routes to survival

Expression of genes alters in response to stress and understanding which genes are perturbed can often provide clues about how a bacteria is surviving\. This can for example help understand the mechanisms of action of, and resistance to antibiotics\. This session will explore responses of *E\. coli* gene expression to meropenem exposure\.

*Schedule*

- Expression analysis introduction – Greg Wickham and Muhammed Yasir\. 9\.30\-9\.45
- RNAseq analysis – 9\.45\-12
- Lunch – 12\-1
- RNAseq analysis continued – 1\-close

Wednesday

No scheduled sessions – time for reinforcement, reflection and research

### Thursday - Transposon mutagenesis

Transposon mutagenesis is a massively parallel method to study the functions of genes in a condition\. Libraries of mutants with transposons randomly inserted within their genomes are constructed and the survival of each mutant assayed in parallel\. There are various of these ‘Tn\-seq’ methods reviewed here:

[https://www\.nature\.com/articles/s41576\-020\-0244\-x](https://www.nature.com/articles/s41576-020-0244-x) 

We will concentrate on TraDIS\-Xpress, a variation of these methods developed at the Quadram Institute: [https://genome\.cshlp\.org/content/30/2/239](https://genome.cshlp.org/content/30/2/239) 

You will be given datasets and instruction on how to analyse them using the QuaTraDIS pipeline: [https://github\.com/quadram\-institute\-bioscience/QuaTradis](https://github.com/quadram-institute-bioscience/QuaTradis) 

*Schedule*

- TraDIS – what is it for, how does it work and why should you care? Emma Holden 9\.30\-10\.
- Tutorial – responses of *E\. coli* to meropenem\. Muhammed Yasir and Emma Holden\. 10\-12\.
- Lunch
- Pm – working through the TraDIS pipeline in groups

### Friday – TraDIS II and wrap up

*Schedule*

- 9\.30\-10\.30 – TraDIS continued\- what worked, what does it tell you? Emma Holden, Muhammed Yasir
- 10\.30\-11\.30 – summary of the week, what have we learned and how can we integrate data? Mark Webber
- 11\.30\-12\. Break 
- 1\- Christmas lunch at Namaste Village

