---
title: Day4
---

# Bioinformatics at the CLI: it's your turn

> Today you will be working on your own on a small project, putting together all the things you learned so far.


## Bioinformatics task

We will map short reads against a reference genome.
Read the tasks first, as this is not a step-by-step tutorial.
You are **not** expected to finish all the tasks, take your time!


### Preliminary steps 

Finding the files needed and gathering the tools using conda.

1. :mag: **Locate the reads**. In our 
["learn_bash" course repository](https://github.com/telatin/learn_bash)
there is a directory containing a set of *paired end* FASTQ files.
[Paired-end files](https://www.illumina.com/science/technology/next-generation-sequencing/plan-experiments/paired-end-vs-single-read.html)
are produced sequencing both ends of the DNA fragments in a library.
It is common for FASTQ files to have the extension `.fastq.gz` (or `.fq.gz`),
as they are usually compressed with gzip.
:bulb: use the `find` command to locate all the FASTQ files in the repository. You should find a single directory containing all of them.
  * :question: How many reads are there in each FASTQ dataset?
  * :question: Do the two FASTQ of each paired-end dataset have the same number of reads?
  * :question: What is the *read length* of each FASTQ dataset?

2. :book: **Reference genome**. In the `./phage/` subdirectory of the repository, 
you will find the files coming from the reference genome sequence and its annotation.
Common extensions for FASTA reference are `.fa`, `.fasta`, `.fna`. We will focus on the file
with the genome, not the proteins/coding sequences.
The annotation is supplied in multiple formats, we will use the `.gff` annotation.

3. :gear: **Gathering the tools**. Create a new conda environment (using mamba) with the following tools. Call the environment "phage":
  
   * *seqfu* for general FASTA/FASTQ manipulations
   * *fastp* for quality filtering
   * *fastqc* for a quality report
   * *bwa* for sequence alignment
   * *samtools* for manipulating SAM/BAM files
   * *skesa* for _de novo_ assembly
  
  * :bulb: For your records, export the new environment to a file called `phage_env.yaml`.

4. :microscope: **Inspect the reference**. Before starting your analyses, check the length and number of sequences in the refernce FASTA file.
You can also open both the reference genome and its annotation using
[IGV](https://telatin.github.io/microbiome-bioinformatics/IGV/) to have a look. 

:warning: From now on we will use **sample2**, but you can repeat the same steps for
other samples if you want.

### Running FastQC (optional)

[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/ ) is a tool for quality control of high-throughput sequencing data. It will summarise
several information about the quality of the reads, and will produce a report in HTML format. It's important to remember that warnings and errors in the report are not necessarily bad, but they can be indicative of problems with the data.

1. Create a new directory called `fastqc`, placed inside the `phage` directory.
2. Run FastQC (you can feed more than one Fastq file at the time):  
```bash
fastqc -o {output_directory} {input_fastq}
```

:question: what files have been created in the output directory?

:information_source: if you are running the analysis locally, open the HTML file with a browser. If you are running on a remote server, you can see the report of 
[Sample2_R1](https://mmbdtp.github.io/fastqc/sample2_R1_fastqc.html) and 
[Sample2_R2](https://mmbdtp.github.io/fastqc/sample2_R2_fastqc.html).

### Mapping (and beyond)

Here we will try to map the reads of one sample against the reference genome.

:warning: We will use **sample2**.

1. :card_index_dividers: **Index the reference genome**. Use `bwa index` to make the reference index mappable by *bwa mem*.
   * :information_source: You can skip this step if you have already done it in a previous exercise.
   * :question: Did the process create new files? Check with `ls`.
   
2. :card_index_dividers: **Map the reads**. Use `bwa mem` to map the reads against the reference genome (you will first need to index the genome). The output from *bwa* is text in SAM format. We are also interested in its sorted BAM output. Using pipes is a good way to do this. 

  * :question: How many header lines and how many body lines are there in the SAM file?

3. :package: **From SAM to BAM**. Use `samtools view` to convert the SAM file to BAM format, and use `samtools sort` to sort the BAM file by coordinates. You can use pipes to do this in a single command
(see [tutorial](https://telatin.github.io/microbiome-bioinformatics/Bash-SAM/)).

4. :mag: **Inspect the BAM files using IGV**. Can you have a feeling of the coverage and differences between the sequenced sample and the reference?

---

### (optional) *De novo* assembly 

> Shotgun sequencing is a laboratory technique for determining the DNA sequence of an organismâ€™s genome. The method involves randomly breaking up the genome into small DNA fragments that are sequenced individually. A computer program looks for overlaps in the DNA sequences, using them to reassemble the fragments in their correct order to reconstitute the genome. ([source](https://www.genome.gov/genetics-glossary/Shotgun-Sequencing))


1. :book: Go to the repository of the [Skesa assembler](https://github.com/ncbi/SKESA#skesa---strategic-k-mer-extension-for-scrupulous-assemblies) and read its documentation and limit to essential parameters:
  * how to supply the input reads in FASTQ format
  * where the output file (assembly) will be stored
  * See the *Example of an assembly that uses separate fastq files for each mate* in the program's docs
2. :rocket: **Perform the assembly**. Use *skesa* to assemble the reads. 
3. :mag: **Inspect the assembly**.
  * :question: How many contigs are there?
  * :question: What is the length of the longest contig? 
  * :question: What is the [N50](https://www.molecularecologist.com/2017/03/29/whats-n50/)?

:star: (optional) try to align the assembly against the reference and see the output using IGV


--- 

![Something different]({{site.baseurl}}/img/different.png)

## A completely different task

This is a bonus track you can open under the tree during the festive season.
It's a non-bioinformatics task.

1. Watch an [interesting video](https://www.youtube.com/watch?v=fCn8zs912OE)
2. Try to see how we can takle the problem with UNIX, and programming languages (Python, R, Perl, etc)

Track: [Word distribution: theory and some exercises](https://telatin.github.io/microbiome-bioinformatics/Words-distribution/)



